modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -2806,4 +2839,16 @@ private void markFinished(UUID txId, TxState txState, @Nullable HybridTimestamp
                 ? null
                 : new TxStateMeta(txState, old.txCoordinatorId(), txState == COMMITED ? commitTimestamp : null));
     }
+
+    // TODO: https://issues.apache.org/jira/browse/IGNITE-20124 tmp
+    private static <T extends Comparable<T>> void updateTrackerIgnoringTrackerClosedException(
```

Comment:
Why we do not update safte time on the primary replica (only ins RAFT state machine), but we do it in your patch?